x/upgrade/module.go

```diff
@@ -101,7 +101,8 @@ func (am AppModule) RegisterServices(cfg module.Configurator) {
 }
 
 // InitGenesis is ignored, no sense in serializing future upgrades
-func (am AppModule) InitGenesis(_ sdk.Context, _ codec.JSONMarshaler, _ json.RawMessage) []abci.ValidatorUpdate {
+func (am AppModule) InitGenesis(ctx sdk.Context, _ codec.JSONMarshaler, _ json.RawMessage) []abci.ValidatorUpdate {
+	InitChainer(am.keeper, ctx)
```

Comment:
@tyler I think i might have been mistaken as to when we should call `keeper.SetConsensusVersions`. We actually don't need to call it in InitChain, and leave it blank on a fresh change, as per https://github.com/cosmos/cosmos-sdk/pull/8504#issuecomment-784452379, I believe it should be called AFTER the handler has run (so [here](https://github.com/cosmos/cosmos-sdk/pull/8743/files#diff-33ff09992f1a3583e2aa8eb932c62c833d52e446488070ed958eb672eed8da3eR241)), to say "ok I've ran the migration scripts, now I'll update the store with the newest module ConsensusVersions'.

I think the ultimate E2E test would be with 2 binaries, but since that's hard, let's try to simulate that in a test here:
- create a mock `module.VersionManager`, say with `bank` set to 1, and everything else set to latest ConsensusVersion.
- `upgradeKeeper.SetModuleManager(mockVersionMgr)`. This simulates saving the "old" modules version to state.
- add an upgrade handler to keeper.
- call ApplyUpgrade on keeper
- after ApplyUpgrade, call `keeper.GetConsensusVersions`, test that the return value's `bank` key should be 2, not 1.

Reply:
If it's not set in IntiChain then how do initial versions get set?? I don't see a good alternative and it's not a safe assumption that all chains start at module version 1

Reply:
I think we need to add this back

Reply:
[InitGenesis added back](https://github.com/cosmos/cosmos-sdk/pull/8743/commits/027740dea5e3770070111f84e68e7f71b6531fde)